sdk/include/reactos/rddm/rxgkinterface.h

```diff
@@ -0,0 +1,411 @@
+DXGADAPTER_WAITFORVERTICALBLANKEVENT(_In_ const D3DKMT_WAITFORVERTICALBLANKEVENT* unnamedParam1);
+
+typedef DXGADAPTER_WAITFORVERTICALBLANKEVENT *PDXGADAPTER_WAITFORVERTICALBLANKEVENT;
+
+typedef
+NTSTATUS
+DXGADAPTER_SIGNALSYNCHRONIZATIONOBJECT(_In_ const D3DKMT_SIGNALSYNCHRONIZATIONOBJECT* unnamedParam1);
+
+typedef DXGADAPTER_SIGNALSYNCHRONIZATIONOBJECT *PDXGADAPTER_SIGNALSYNCHRONIZATIONOBJECT;
+
+typedef
+NTSTATUS
+DXGADAPTER_QUERYRESOURCEINFO(_Inout_ D3DKMT_QUERYRESOURCEINFO* unnamedParam1);
+
+typedef DXGADAPTER_QUERYRESOURCEINFO *PDXGADAPTER_QUERYRESOURCEINFO;
+
+typedef
+NTSTATUS
+DXGADAPTER_GETPRESENTHISTORY(_Inout_ D3DKMT_GETPRESENTHISTORY* unnamedParam1);
+
+typedef DXGADAPTER_GETPRESENTHISTORY *PDXGADAPTER_GETPRESENTHISTORY;
+
+typedef
+NTSTATUS
+DXGADAPTER_SETALLOCATIONPRIORITY(_In_ const D3DKMT_SETALLOCATIONPRIORITY* unnamedParam1);
+
+typedef DXGADAPTER_SETALLOCATIONPRIORITY *PDXGADAPTER_SETALLOCATIONPRIORITY;
+
+typedef
+NTSTATUS
+DXGADAPTER_INVALIDATEACTIVEVIDPN(_In_ const D3DKMT_INVALIDATEACTIVEVIDPN* unnamedParam1);
+
+typedef DXGADAPTER_INVALIDATEACTIVEVIDPN *PDXGADAPTER_INVALIDATEACTIVEVIDPN;
+
+typedef
+NTSTATUS
+DXGADAPTER_GETSHAREDPRIMARYHANDLE(_In_ D3DKMT_GETSHAREDPRIMARYHANDLE* unnamedParam1);
+
+typedef DXGADAPTER_GETSHAREDPRIMARYHANDLE *PDXGADAPTER_GETSHAREDPRIMARYHANDLE;
+
+typedef
+NTSTATUS
+DXGADAPTER_LOCK(_In_ D3DKMT_LOCK* unnamedParam1);
+
+typedef DXGADAPTER_LOCK *PDXGADAPTER_LOCK;
+
+typedef
+NTSTATUS
+DXGADAPTER_POLLDISPLAYCHILDREN(_In_ const D3DKMT_POLLDISPLAYCHILDREN* unnamedParam1);
+
+typedef DXGADAPTER_POLLDISPLAYCHILDREN *PDXGADAPTER_POLLDISPLAYCHILDREN;
+
+typedef
+NTSTATUS
+DXGADAPTER_OPENRESOURCE(_Inout_ D3DKMT_OPENRESOURCE* unnamedParam1);
+
+typedef DXGADAPTER_OPENRESOURCE *PDXGADAPTER_OPENRESOURCE;
+
+typedef
+NTSTATUS
+DXGADAPTER_PRESENT(_In_ D3DKMT_PRESENT* unnamedParam1);
+
+typedef DXGADAPTER_PRESENT *PDXGADAPTER_PRESENT;
+
+typedef
+NTSTATUS
+DXGADAPTER_QUERYADAPTERINFO(_Inout_ const D3DKMT_QUERYADAPTERINFO* unnamedParam1);
+
+typedef DXGADAPTER_QUERYADAPTERINFO *PDXGADAPTER_QUERYADAPTERINFO;
+
+typedef
+NTSTATUS
+DXGADAPTER_QUERYALLOCATIONRESIDENCY(_In_ const D3DKMT_QUERYALLOCATIONRESIDENCY* unnamedParam1);
+
+typedef DXGADAPTER_QUERYALLOCATIONRESIDENCY *PDXGADAPTER_QUERYALLOCATIONRESIDENCY;
+
+typedef
+NTSTATUS
+DXGADAPTER_QUERYSTATISTICS(_Inout_ const D3DKMT_QUERYSTATISTICS* unnamedParam1);
+
+typedef DXGADAPTER_QUERYSTATISTICS *PDXGADAPTER_QUERYSTATISTICS;
+
+typedef
+NTSTATUS
+DXGADAPTER_RELEASEPROCESSVIDPNSOURCEOWNERS(_In_ HANDLE unnamedParam1);
+
+typedef DXGADAPTER_RELEASEPROCESSVIDPNSOURCEOWNERS *PDXGADAPTER_RELEASEPROCESSVIDPNSOURCEOWNERS;
+
+typedef
+NTSTATUS
+DXGADAPTER_RENDER(_In_ D3DKMT_RENDER* unnamedParam1);
+
+typedef DXGADAPTER_RENDER *PDXGADAPTER_RENDER;
+
+typedef
+NTSTATUS
+DXGADAPTER_SETCONTEXTSCHEDULINGPRIORITY(_In_ const D3DKMT_SETCONTEXTSCHEDULINGPRIORITY* unnamedParam1);
+
+typedef DXGADAPTER_SETCONTEXTSCHEDULINGPRIORITY *PDXGADAPTER_SETCONTEXTSCHEDULINGPRIORITY;
```

Comment:
@DarkFire01 Question: Is having two typedefs for each function types, instead of just one as in e.g.
https://git.reactos.org/?p=reactos.git;a=blob;f=sdk/include/ddk/video.h;h=3649f8c60a909fc5619fd05fdd0f2d69fcace68e;hb=HEAD#l161
(for example the last typedef could be:
```c
typedef NTSTATUS
(*PDXGADAPTER_SETCONTEXTSCHEDULINGPRIORITY)(_In_ const D3DKMT_SETCONTEXTSCHEDULINGPRIORITY* unnamedParam1);
```
),
for compatibility purposes with other official MS headers that specify both types?

Reply:
Yeah I've seen it kinda thrown around back and fourth.